src/SprykerEco/Zed/Payone/Communication/Controller/TransactionController.php

```diff
@@ -62,42 +61,38 @@ public function statusUpdateAction(Request $request)
         $payoneTransactionStatusUpdateTransfer = new PayoneTransactionStatusUpdateTransfer();
         $payoneTransactionStatusUpdateTransfer->fromArray($dataArray);
 
-        $response = $this->getFacade()->processTransactionStatusUpdate($payoneTransactionStatusUpdateTransfer);
+        $payoneTransactionStatusUpdateTransfer = $this->getFacade()->processTransactionStatusUpdate($payoneTransactionStatusUpdateTransfer);
 
-        $transactionId = $payoneTransactionStatusUpdateTransfer->getTxid();
-        $this->triggerEventsOnSuccess($response, $transactionId, $dataArray);
+        $this->triggerEventsOnSuccess($payoneTransactionStatusUpdateTransfer);
 
-        $callback = function () use ($response) {
-            echo $response;
+        $callback = function () use ($payoneTransactionStatusUpdateTransfer) {
+            echo $payoneTransactionStatusUpdateTransfer->getResponse();
         };
 
         return $this->streamedResponse($callback);
     }
 
     /**
-     * @param \SprykerEco\Zed\Payone\Business\Api\TransactionStatus\TransactionStatusResponse $response
-     * @param int $transactionId
-     * @param array $dataArray
+     * @param \Generated\Shared\Transfer\PayoneTransactionStatusUpdateTransfer $payoneTransactionStatusUpdateTransfer
      *
      * @return void
      */
-    protected function triggerEventsOnSuccess(TransactionStatusResponse $response, $transactionId, array $dataArray)
+    protected function triggerEventsOnSuccess(PayoneTransactionStatusUpdateTransfer $payoneTransactionStatusUpdateTransfer)
     {
-        if (!$response->isSuccess()) {
+        if (!$payoneTransactionStatusUpdateTransfer->getIsSuccess()) {
             return;
         }
 
         //TODO: Refactor as per CD-380
```

Comment:
What about this TODO ?

Reply:
Should be removed and can be added to https://release.spryker.com/module-cleanups